app/src/main/java/ch/epfl/sdp/blindly/match/MatchActivity.kt

```diff
@@ -0,0 +1,149 @@
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        profiles = intent.getParcelableArrayListExtra("EXTRA_MATCH_PROFILES")!!
+        setContentView(R.layout.activity_match)
+        setupCardStackView()
+        setupButton()
+    }
+
+    override fun onBackPressed() {
+        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
+            drawerLayout.closeDrawers()
+        } else {
+            super.onBackPressed()
+        }
+    }
+
+    override fun onCardDragging(direction: Direction, ratio: Float) {
+    }
+
+    /**
+     * here we have to save if the profile is liked or not for messaging him/her after
+     *
+     * @param direction
+     */
+    override fun onCardSwiped(direction: Direction) {
+    }
+
+    override fun onCardRewound() {
+    }
+
+    override fun onCardCanceled() {
+    }
+
+    override fun onCardAppeared(view: View, position: Int) {
+        val textView = view.findViewById<TextView>(R.id.item_name)
+        Log.d("CardStackView", "onCardAppeared: ($position) ${textView.text}")
+    }
+
+    override fun onCardDisappeared(view: View, position: Int) {
+        val textView = view.findViewById<TextView>(R.id.item_name)
+        Log.d("CardStackView", "onCardDisappeared: ($position) ${textView.text}")
+    }
+
+
+    private fun setupCardStackView() {
+        initialize()
+    }
+
+    private fun setupButton() {
+        setupLikeButton()
+        setupRewindButton()
+        setupSkipButton()
+    }
+
+    private fun setupLikeButton() {
+        val like = findViewById<View>(R.id.like_button)
+        like.setOnClickListener {
+            val setting = SwipeAnimationSetting.Builder()
+                .setDirection(Direction.Right)
+                .setDuration(Duration.Normal.duration)
+                .setInterpolator(AccelerateInterpolator())
+                .build()
+            manager.setSwipeAnimationSetting(setting)
+            cardStackView.swipe()
+        }
+    }
+
+    private fun setupSkipButton() {
+        val skip = findViewById<View>(R.id.skip_button)
+        skip.setOnClickListener {
+            val setting = SwipeAnimationSetting.Builder()
+                .setDirection(Direction.Left)
+                .setDuration(Duration.Normal.duration)
+                .setInterpolator(AccelerateInterpolator())
+                .build()
+            manager.setSwipeAnimationSetting(setting)
+            cardStackView.swipe()
+        }
+    }
+
+    private fun setupRewindButton() {
+        val rewind = findViewById<View>(R.id.rewind_button)
+        rewind.setOnClickListener {
+            val setting = RewindAnimationSetting.Builder()
+                .setDirection(Direction.Bottom)
+                .setDuration(Duration.Normal.duration)
+                .setInterpolator(DecelerateInterpolator())
+                .build()
+            manager.setRewindAnimationSetting(setting)
+            cardStackView.rewind()
+        }
+    }
```

Comment:
I think you can refactor those two methods into one using a boolean or something

Reply:
I think it might be clearer if we keep them like this, even if the functions are similar.

Reply:
I'm not sure it would be less clear, with a method setupButton and a boolean isSkipButton or isLikeButton I think it would be better. The methods are more than similar I think